ios/RNVitals.m

```diff
@@ -0,0 +1,120 @@
+
+#import "RNVitals.h"
+#import <mach/mach.h>
+#import <mach/mach_host.h>
+
+// Used to send events to JS
+#if __has_include(<React/RCTBridge.h>)
+#import <React/RCTBridge.h>
+#import <React/RCTEventDispatcher.h>
+#elif __has_include("RCTBridge.h")
+#import "RCTBridge.h"
+#import "RCTEventDispatcher.h"
+#else
+#import "React/RCTBridge.h"
+#import "React/RCTEventDispatcher.h"
+#endif
+
+@implementation RNVitals
+
+@synthesize bridge = _bridge;
+
+static NSString * const LOW_MEMORY = @"LOW_MEMORY";
+
+- (instancetype)init
+{
+    if ((self = [super init])) {
+        [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(applicationDidReceiveMemoryWarning:) name:UIApplicationDidReceiveMemoryWarningNotification object:nil];
+    }
+
+    return self;
+
+}
+
+- (void)dealloc
+{
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
++ (NSDictionary *)getMemoryInfo
+{
+    struct mach_task_basic_info info;
+    mach_msg_type_number_t size = sizeof(info);
+    kern_return_t kerr = task_info(mach_task_self(),
+                                   MACH_TASK_BASIC_INFO,
+                                   (task_info_t)&info,
+                                   &size);
+
+    if (kerr != KERN_SUCCESS) {
+        return nil;
+    }
+
+    return @{
+             @"total": [NSNumber numberWithUnsignedLongLong:[NSProcessInfo processInfo].physicalMemory],
+             @"used": [NSNumber numberWithUnsignedLongLong:info.resident_size],
+             @"free": [NSNumber numberWithUnsignedLongLong:info.virtual_size]
+             };
+}
+
+- (void)applicationDidReceiveMemoryWarning:(NSNotification *)notification
+{
+    NSDictionary *memoryInfo = [RNVitals getMemoryInfo];
+    if (memoryInfo == nil)
+    {
+        [_bridge.eventDispatcher sendDeviceEventWithName:LOW_MEMORY
+                                                    body:@"An error happened while getting memory logs"];
```

Comment:
The subscriber shouldn't have to expect and handle two different body formats. This is like an API: how would you handle responding to a request with an error rather than a successful response?

In this scenario, let the response be empty. Does `eventDispatcher` provide a method to send an event without a body? If not, can you see if you can send `NSNull`? We'll want to double check that sending `NSNull` is safe

Reply:
So sending `nill` will return `undefined` on the JS side. Will that work?

Reply:
Sure. Then the JS side can do a simple existence check